### ISSUE TITLE
[FR] Official support for InnoDB

### ISSUE THREAD

**Issue by madchap** [etianen/django-watson#273]
Hi,

So this is more like a feature request time of issue.

I know there's been a couple (closed) issues, yet supporting InnoDB looks like a natural and needed move for Django-watson if it is to continue being useful for many people.

**Comment by etianen** [comment]
I agree. Unfortunately, I don't have the time to implement it myself. I would happily review a MR.

**Comment by tari** [comment]
I've been using Watson with InnoDB on MariaDB for a while, just by manually changing the engine on the `watson_searchentry` table to be InnoDB. My application doesn't push it very hard so it's possible there are some features that wouldn't work, but it seems to behave correctly.

There is one issue I've found that's more like a database bug, though. MySQL documentation says that `@` isn't supported at all by InnoDB fulltext search `IN BOOLEAN MODE` (and MariaDB seems to inherit this behavior, but doesn't document it):

> InnoDB full-text search does not support the use of the @ symbol in boolean full-text searches. The @ symbol is reserved for use by the `@distance` proximity search operator.

The consequence of doing so is a very confusing syntax error (like https://stackoverflow.com/q/25088183/2658436), and https://bugs.mysql.com/bug.php?id=74042 basically says the developers don't care about enabling searches for `@`. Ole Martin's workaround of replacing `@` with ` ` seems like the best you can do on InnoDB, so official InnoDB support in Watson would probably want to do that at the least.

**Comment by etianen** [comment]
You sound like an excellent person to formulate a PR! ❤️
